doc/dev/ai_prompt_workflow.md

```diff
@@ -0,0 +1,14 @@
+# AI Prompt Workflow
```

Comment:
Will we be adding more AI docs of this nature? I wonder if we should start a doc/dev/ai directory?